src/load.jl

```diff
@@ -0,0 +1,217 @@
+    return push_call!(tape, add, args...)
+end
+
+
+# function load_node!(tape::Tape, ::OpConfig{:ONNX, :Mul}, args::VarVec, attrs::AttrDict)
+#     return push_call!(tape, mul, args...)
+# end
+
+
+# function load_node!(tape::Tape, ::OpConfig{:ONNX, :Relu}, args::VarVec, attrs::AttrDict)
+#     return push_call!(tape, relu, args[1])
+# end
+
+
+# function load_node!(tape::Tape, ::OpConfig{:ONNX, :MaxPool}, args::VarVec, attrs::AttrDict)
+#     _,k,p,s,_ = akpsd(attrs)
+#     return push_call!(tape, maxpool, args[1], k; pad=p, stride=s)
+# end
+
+
+# function load_node!(tape::Tape, ::OpConfig{:ONNX, :BatchNormalization}, args::VarVec, attrs::AttrDict)
+#     ϵ = get(attrs, :epsilon, 1f-5)
+#     momentum = get(attrs, :momentum, 9f-1)
+#     training_mode = Bool(get(attrs, :training_mode, 0))
+#     res = push_call!(tape, batch_norm, args..., ϵ, momentum, training_mode)
+#     if training_mode
+#         y = push_call!(tape, getfield, 1)
+#         μ_new = push_call!(tape, getfield, 2)
+#         σ²_new = push_call!(tape, getfield, 3)
+#         return y, μ_new, σ²_new
+#     else
+#         return res
+#     end
+# end
+
+
+# function load_node!(tape::Tape, ::OpConfig{:ONNX, :GlobalAveragePool}, args::VarVec, attrs::AttrDict)
+#     return push_call!(tape, global_average_pool, args...)
+# end
+
+
+###############################################################################
+#                                    API                                      #
+###############################################################################
+
+
+"""
+    load(io::IO, model_args...; backends=[:ONNX], exec::Bool=true)
+    load(filename::String, model_args...; backends=[:ONNX], exec::Bool=true)
+
+Load an ONNX model as a Ghost.Tape. The way a particular ONNX node is deserialized is
+controlled by methods of [load_node!](@ref) dispatched by backend and node's op_type.
+
+`backends` parameter can be used to customize the loading process.
+
+`exec` parameter instructs the loader to execute every added operation just after
+the addition, making the debugging easier. Default is `true`.
+
+See also: [save!](@ref)
+"""
+function load(io::IO, model_args...; backends=[:ONNX], exec::Bool=true)
+    onnx_model = readproto(io, ModelProto());
+    g = onnx_model.graph;
+    tape = Tape(ONNXCtx(backends; exec=exec))
+    # create map of initializers
+    init_vals = Dict{String, Any}()
+    for init in g.initializer
+        # TODO: consider non-array inputs
+        init_vals[init.name] = array(init)
+    end
+    # load inputs
+    arg_idx = 1
+    for inp in g.input
+        val = get(init_vals, inp.name, missing)
+        if val === missing && exec == true
+            val = model_args[arg_idx]
+            arg_idx += 1
+        end
+        v = push!(tape, Input(val))
+        tape.c.name2var[inp.name] = v
+    end
+    # load nodes
+    op_configs = [meth.sig.parameters[3] for meth in methods(load_node!).ms]
```

Comment:
Looking for method signatures look a bit scary to me. Would it be better to have a generic method which returns a sentinel value (e.g. `missing` or `NotImplemented()`) for the loop below?